consensus/scp/src/quorum_set.rs

```diff
@@ -39,18 +29,68 @@ pub enum QuorumSetMember<ID: GenericNodeId> {
     InnerSet(QuorumSet<ID>),
 }
 
-impl<
-        ID: GenericNodeId
-            + Clone
-            + Debug
-            + Display
-            + Serialize
-            + DeserializeOwned
-            + Eq
-            + PartialEq
-            + Hash,
-    > QuorumSet<ID>
-{
+impl<ID: GenericNodeId> PartialEq for QuorumSetMember<ID> {
+    fn eq(&self, other: &QuorumSetMember<ID>) -> bool {
+        match self {
+            QuorumSetMember::Node(self_node) => match other {
+                QuorumSetMember::Node(other_node) => self_node == other_node,
+                _ => false,
+            },
+            QuorumSetMember::InnerSet(self_qs) => match other {
+                QuorumSetMember::InnerSet(other_qs) => self_qs == other_qs,
+                _ => false,
+            },
+        }
+    }
+}
+impl<ID: GenericNodeId> Eq for QuorumSetMember<ID> {}
+
+impl<ID: GenericNodeId> Hash for QuorumSetMember<ID> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self {
+            QuorumSetMember::Node(self_node) => self_node.hash(state),
+            QuorumSetMember::InnerSet(self_qs) => self_qs.hash(state),
+        }
+    }
+}
+
+/// The quorum set defining the trusted set of peers.
+#[derive(Clone, Debug, Ord, PartialOrd, Serialize, Deserialize, Digestible)]
+pub struct QuorumSet<ID: GenericNodeId = NodeID> {
+    /// Threshold (how many members do we need to reach quorum).
+    pub threshold: u32,
+
+    /// Members.
+    pub members: Vec<QuorumSetMember<ID>>,
```

Comment:
Have you looked into whether this can be changed to a `HashSet`? That would technically be more correct since it would also prevent duplicate items in the set.
I expect it shouldn't be too difficult since the constructor methods could continue to accept `Vec`. However, I am not sure what implications it has over `Hash`.

Reply:
I tried this and it seemed like it would be a more substantial change with complexity related to the no_std HasherBuilder used in mc_common. I think it could be worth changing the entire scp crate to use `std::HashSet` and modifying the `QuorumSet` definition to use HashSet instead of Vec in a future PR.

Reply:
scp doesn't need to be no-std.

Reply:
> scp doesn't need to be no-std.

Agreed... we've just fallen into a pattern of using `mc_common::{HashSet, HashMap}` everywhere as a default. I'll open a ticket to consider changing the scp crate specifically to std::HashSet.

Reply:
https://mobilecoin.atlassian.net/browse/MC-1594 -- should we address this in this PR or in a future PR?

Reply:
looking a bit more closely at this, we should prob. use BTreeSets given how `findQuorumHelper` and `findBlockingSetHelper` want to use recursion with a consistent ordering.

Reply:
unless we want to also rewrite the predicate helpers in a way that doesn't use vector slices recursively, I don't think we're going to gain much by switching members to a set.